### PR TITLE
de: Translation until chapter 9.1

### PR DESCRIPTION
**Pull request by Throvn**
Thank you so much @mgeisler for figuring out how to solve the huge diff problem.
@ronaldfw it would be great if you could do your magic and put my german skills to the test once more :)

### CODE REVIEW

File: po/de.po [hunk 1]
```diff
@@ -3757,48 +3749,44 @@ msgid ""
...
-"-Methode und ähnlich hat das `Into<T>`-Merkmal eine einzige `into()`-"
-"Methode.\n"
-"Durch die Implementierung dieser Merkmale drückt ein Typ aus, in was er "
-"umgewandelt werden kann\n"
-"ein anderer Typ."
+"Alle Rust-Integer-Typen implementieren die [`From<T>`][1]- und [`Into<T>`]"
+"[2]-Merkmale (traits), \n"
+"um zwischen ihnen umzuwandeln. Das `From<T>`-Merkmal (trait) hat eine "
+"einzige `from()`\n"
+"-Methode und ähnlicherweise hat das `Into<T>`-Merkmal auch nur eine einzige "
```
Comment (ronaldfw) [inline po/de.po:3756]: "ähnlicherweise" sounds strange and according to Duden, it's actually not a German word. You could write
"in ähnlicher Weise", but I would simply shorten this to "ebenso" or "genauso". These latter sound more natural to me.

File: po/de.po [hunk 2]
```diff
@@ -3757,48 +3749,44 @@ msgid ""
...
 msgid ""
 "The same applies for your own `From` implementations for your own types, so "
 "it is\n"
 "sufficient to only implement `From` to get a respective `Into` "
 "implementation automatically."
 msgstr ""
-"Dasselbe gilt für Ihre eigenen `From`-Implementierungen für Ihre eigenen "
-"Typen, so ist es\n"
-"Es reicht aus, nur `From` zu implementieren, um automatisch eine "
+"Dasselbe gilt für Deine eigenen `From`-Implementierungen in Deinen eigenen "
```
Comment (ronaldfw) [inline po/de.po:3784]:
```suggestion
"Dasselbe gilt für `From`-Implementierungen für Deine eigenen "
```

File: po/de.po [hunk 3]
```diff
@@ -3837,13 +3826,12 @@ msgid ""
...
 
 #: src/exercises/day-1/for-loops.md:9
-#, fuzzy
 msgid ""
 "You can print such an array by asking for its debug representation with `{:?}"
 "`:"
 msgstr ""
-"Sie können ein solches Array drucken, indem Sie mit `{:?}` nach seiner Debug-"
-"Darstellung fragen:"
+"Du können ein solches Array drucken (print), indem Du mit `{:?}` nach der "
```
Comment (ronaldfw) [inline po/de.po:3833]:
```suggestion
"Du kannst ein solches Array drucken (print), indem Du mit `{:?}` nach der "
```

File: po/de.po [hunk 4]
```diff
@@ -3954,58 +3937,51 @@ msgid ""
...
 "slice-of-slices. Why or why not?"
 msgstr ""
-"Könnten Sie `&[i32]`-Slices anstelle von fest codierten 3 × 3-Matrizen für "
-"Ihre verwenden\n"
-"Argument- und Rückgabetypen? Etwas wie `&[&[i32]]` für ein "
-"zweidimensionales\n"
-"Scheibe-von-Scheiben. Warum oder warum nicht?"
+"Könntest Du `&[i32]`-Slices anstelle von fest codierten 3 × 3-Matrizen als\n"
+"Argument- und Rückgabetypen verwenden? Oder etwas wie `&[&[i32]]` für ein "
+"zweidimensionales Slice? \n"
```
Comment (ronaldfw) [inline po/de.po:3951]: The German translation of the Rust Programming Language book uses "Anteilstyp" as translation for Rust; https://rust-lang-de.github.io/rustbook-de/ch04-03-slices.html

File: po/de.po [hunk 5]
```diff
@@ -3954,58 +3937,51 @@ msgid ""
...
 "Could you use `&[i32]` slices instead of hard-coded 3 × 3 matrices for your\n"
 "argument and return types? Something like `&[&[i32]]` for a two-dimensional\n"
 "slice-of-slices. Why or why not?"
 msgstr ""
-"Könnten Sie `&[i32]`-Slices anstelle von fest codierten 3 × 3-Matrizen für "
-"Ihre verwenden\n"
-"Argument- und Rückgabetypen? Etwas wie `&[&[i32]]` für ein "
-"zweidimensionales\n"
-"Scheibe-von-Scheiben. Warum oder warum nicht?"
+"Könntest Du `&[i32]`-Slices anstelle von fest codierten 3 × 3-Matrizen als\n"
```
Comment (ronaldfw) [inline po/de.po:3949]:
```suggestion
"Könntest Du den `&[i32]`-Anteilstyp anstelle von fest codierten 3 × 3-Matrizen als\n"
```

File: po/de.po [hunk 6]
```diff
@@ -3954,58 +3937,51 @@ msgid ""
...
+"Argument- und Rückgabetypen verwenden? Oder etwas wie `&[&[i32]]` für ein "
```
Comment (ronaldfw) [inline po/de.po:3950]:
```suggestion
"Argument- und Rückgabetypen verwenden? Oder etwas wie `&[&[i32]]` für einen "
```

File: po/de.po (hunk 4 above)
Comment (ronaldfw) [inline po/de.po:3951]:
```suggestion
"zweidimensionalen Anteilstyp? \n"
```

File: po/de.po [hunk 7]
```diff
@@ -4086,22 +4058,21 @@ msgid ""
...
 "Es ist sehr wichtig zu betonen, dass auf diese Weise deklarierte Variablen "
-"nicht von einer Art dynamischem \"beliebigen Typ\" sind, der dies kann\n"
-"irgendwelche Daten halten. Der durch eine solche Deklaration erzeugte "
+"**nicht** von einem dynamischem \"beliebigen Typ\" sind, der\n"
+"beliebige Daten enthalten kann. Der durch eine solche Deklaration erzeugte "
 "Maschinencode ist identisch mit der expliziten Deklaration eines Typs.\n"
-"Der Compiler erledigt die Arbeit für uns und hilft uns, einen prägnanteren "
-"Code zu schreiben."
+"Der Compiler erledigt lediglich die Arbeit für uns und hilft uns, einen "
+"prägnanteren Code zu schreiben."
```
Comment (ronaldfw) [inline po/de.po:4065]:
```suggestion
"Der Compiler erledigt lediglich die Arbeit für uns und hilft uns "
"prägnanteren Code zu schreiben."
```

Comment (ronaldfw) [reply po/de.po:4065]: Dropping the "einen" sounds cleaner.

File: po/de.po [hunk 8]
```diff
@@ -4086,22 +4058,21 @@ msgid ""
...
 msgid ""
 "The following code tells the compiler to copy into a certain generic "
 "container without the code ever explicitly specifying the contained type, "
 "using `_` as a placeholder:"
 msgstr ""
 "Der folgende Code weist den Compiler an, in einen bestimmten generischen "
-"Container zu kopieren, ohne dass der Code jemals explizit den enthaltenen "
-"Typ angibt, wobei `_` als Platzhalter verwendet wird:"
+"Container zu kopieren, ohne dass jemals der enthaltene Typ explizit "
+"angegeben wird. `_` wird dabei als Platzhalter verwendet:"
```
Comment (ronaldfw) [inline po/de.po:4075]:
```suggestion
"angegeben wird. Der Unterstrich `_` wird dabei als Platzhalter verwendet:"
```

Comment (ronaldfw) [reply po/de.po:4075]: (to avoid starting the sentence with '_')

File: po/de.po [hunk 9]
```diff
@@ -4209,50 +4173,46 @@ msgid ""
...
 #: src/basic-syntax/static-and-const.md:44
-#, fuzzy
 msgid ""
 "We will look at mutating static data in the [chapter on Unsafe Rust](../"
 "unsafe.md)."
 msgstr ""
-"Wir werden uns im [Kapitel über Unsafe Rust](../unsafe.md) mit mutierenden "
-"statischen Daten befassen."
+"Wir werden uns im [Kapitel über unsicheres Rust](../unsafe.md) mit "
+"mutierenden statischen Daten befassen."
```
Comment (ronaldfw) [inline po/de.po:4185]: I think the better translation here is "veränderbaren" instead of "mutierenden". The intention is not that the data mutates by itself, but that we are mutating the data. If you want to translate more directly, I think the following would also work: 

"Wir werden uns im Kapitel über unsicheres Rust mit dem Verändern von statischen Daten befassen."

File: po/de.po [hunk 10]
```diff
@@ -4209,50 +4173,46 @@ msgid ""
...
-#, fuzzy
 msgid ""
 "* Mention that `const` behaves semantically similar to C++'s `constexpr`.\n"
 "* `static`, on the other hand, is much more similar to a `const` or mutable "
 "global variable in C++.\n"
 "* It isn't super common that one would need a runtime evaluated constant, "
 "but it is helpful and safer than using a static."
 msgstr ""
-"* Erwähnen Sie, dass sich `const` semantisch ähnlich wie `constexpr` von C++ "
+"* Erwähne, dass sich `const` semantisch ähnlich wie `constexpr` von C++ "
```
Comment (ronaldfw) [inline po/de.po:4195]:
```suggestion
"* Erwähne, dass sich `const` semantisch ähnlich wie `constexpr` in C++ "
```

File: po/de.po [hunk 11]
```diff
@@ -4287,16 +4246,16 @@ msgid ""
 "locations when shadowing an immutable variable in a scope, even if the type "
 "does not change."
 msgstr ""
-"* Definition: Shadowing unterscheidet sich von Mutation, da nach dem "
-"Shadowing die Speicherplätze beider Variablen gleichzeitig existieren. Beide "
-"sind unter demselben Namen verfügbar, je nachdem, wo Sie ihn im Code "
-"verwenden.\n"
+"* Definition: Verschattung (shadowing) unterscheidet sich von Mutation, da "
+"nach dem Shadowing die Speicherplätze beider Variablen gleichzeitig "
```
Comment (ronaldfw) [inline po/de.po:4250]:
```suggestion
"nach der Verschattung die Speicherplätze beider Variablen gleichzeitig "
```

File: po/de.po [hunk 12]
```diff
@@ -4287,16 +4246,16 @@ msgid ""
...
+"existieren. Beide sind unter demselben Namen verfügbar, je nachdem, wo Sie "
+"ihn im Code verwenden.\n"
```
Comment (ronaldfw) [inline po/de.po:4252]:
```suggestion
"existieren. Beide Speicherplätze sind unter demselben Namen verfügbar, je nachdem, wo Sie "
"die Variablen im Code verwenden.\n"
```

Comment (ronaldfw) [reply po/de.po:4252]: Added the nouns to disambiguate the sentence a bit.

**Review by ronaldfw — APPROVED** [review]
Excellent work. And after the changes with the diff, it's a lot easier to review. Thanks!

File: po/de.po (hunk 4 above)
Comment (Throvn) [reply po/de.po:3951]: The idea was to start with the german translations and then slowly change to the english once as one progresses further, because these are the names you will inevitably use. However, I see that this is confusing and harder to understand because you suddenly have two names for the same thing. I will adhere stricktly to the german names in my following PRs.

File: po/de.po (hunk 9 above)
Comment (Throvn) [reply po/de.po:4185]: Makes sense, I don't care which version we use though, "veränderbaren" or "mit dem Verändern von" is indifferent to me.

File: po/de.po (hunk 11 above)
Comment (Throvn) [reply po/de.po:4250]: How often should we write the english translation in parentethes after the keyword?

Comment (mgeisler) [reply po/de.po]: Hey @Throvn, I don't think we have a standard for this across translations — we've let the different language groups decide this for themselves. If we should give general advice, then we should put it in [STYLE.md](https://github.com/google/comprehensive-rust/blob/main/STYLE.md#translations).

@henrif75, you might be interested in this too.
